### PR TITLE
Velocloud-Junit_Coverage#5918

### PR DESCRIPTION
**Pull request by siddalasha**
Fixes #5918
 Results:
[INFO]
[INFO] Tests run: 44, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO]
[INFO] --- jacoco-maven-plugin:0.8.4:report (post-unit-test) @ velocloud-sdwan-driver ---
[INFO] Loading execution data file C:\project\velocloud-sdwan-driver\target\test-results\coverage\jacoco\jacoco.exec
[INFO] Analyzed bundle 'velocloud-sdwan-driver' with 53 classes
[INFO]
[INFO] --- maven-jar-plugin:3.2.2:jar (default-jar) @ velocloud-sdwan-driver ---
[INFO] Building jar: C:\project\velocloud-sdwan-driver\target\velocloud-sdwan-driver-1.0.1-SNAPSHOT.jar
[INFO]
[INFO] --- spring-boot-maven-plugin:2.7.1:repackage (repackage) @ velocloud-sdwan-driver ---
[INFO] Replacing main artifact with repackaged archive
[INFO]
[INFO] --- sortpom-maven-plugin:2.8.0:sort (default) @ velocloud-sdwan-driver ---
[INFO] Sorting file C:\project\velocloud-sdwan-driver\pom.xml
[INFO] Pom file is already sorted, exiting
[INFO]
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ velocloud-sdwan-driver ---
[INFO] Installing C:\project\velocloud-sdwan-driver\target\velocloud-sdwan-driver-1.0.1-SNAPSHOT.jar to C:\Users\gsiddala\.m2\repository\com\ibm\velocloud-sdwan-driver\velocloud-sdwan-driver\1.0.1-SNAPSHOT\velocloud-sdwan-driver-1.0.1-SNAPSHOT.jar
[INFO] Installing C:\project\velocloud-sdwan-driver\pom.xml to C:\Users\gsiddala\.m2\repository\com\ibm\velocloud-sdwan-driver\velocloud-sdwan-driver\1.0.1-SNAPSHOT\velocloud-sdwan-driver-1.0.1-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  40.528 s
[INFO] Finished at: 2022-09-14T10:32:22+05:30

### CODE REVIEW

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	
+	@Test
+ 	@DisplayName("Testing positive scenario to get EdgeConfigurationStack")
+	public void getEdgeConfigurationStackTest() {
+		
+		ResponseEntity<String> response = new ResponseEntity<String>("", HttpStatus.OK);
+		ArgumentCaptor<HttpEntity> httpEntityArgumentCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+		when(restTemplate.exchange(eq(DESCRIPTORS_API_URL), eq(HttpMethod.POST), httpEntityArgumentCaptor.capture(), 
+				eq(String.class))).thenReturn(response);
+		sdWanDriver.getEdgeConfigurationStack(deploymentLocationProperties, "", "");
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:57]: What is the expectation after calling sdWanDriver.getEdgeConfigurationStack() method?

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	
+	@Test
+	@DisplayName("Testing positive scenario to getURL for LIFECYCLE_CREATE")
+	public void getURLforLIFECYCLE_CREATE () {
+		String url = "http://localhost:8080/api/velocloud/edge/edgeProvision";
+		ResponseEntity<String> response = new ResponseEntity<String>("", HttpStatus.OK);
+		ArgumentCaptor<HttpEntity> httpEntityArgumentCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+		when(restTemplate.exchange(eq(url), eq(HttpMethod.POST), httpEntityArgumentCaptor.capture(), 
+				eq(String.class))).thenReturn(response);
+		sdWanDriver.execute(Constants.LIFECYCLE_CREATE, deploymentLocationProperties, "", "");
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:90]: Where is the getURL() function call?

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	}
+	
+	@Test
+	@DisplayName("Testing positive scenario getURL for LIFECYCLE_DELETE")
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:94]: Is this really a positive scenario?

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	public void getURLforLIFECYCLE_DELETE() {
+		String url = "http://localhost:8080/api/velocloud/edge/deleteEdge";
+		String resp = null;
+		ResponseEntity<String> response = new ResponseEntity<String>(resp, HttpStatus.ACCEPTED);
+		ArgumentCaptor<HttpEntity> httpEntityArgumentCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+		when(restTemplate.exchange(eq(url), eq(HttpMethod.POST), httpEntityArgumentCaptor.capture(), 
+				eq(String.class))).thenReturn(response);
+		assertThrows(SdwanResponseException.class, ()->{
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:102]: Here also I don't see getURL() function call which is your objective.

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+	@Test
+	@DisplayName("Testing positive scenario getURL for LIFECYCLE_OPERATION_DELETE_VCE_STATIC_IP")
+	public void getURLforLIFECYCLE_OPERATION_DELETE_VCE_STATIC_IP() {
+		String url = "http://localhost:8080/api/velocloud/configuration/updateConfigurationModule";
+		String resp = null;
+		ResponseEntity<String> response = new ResponseEntity<String>(resp, HttpStatus.BAD_REQUEST);
+		ArgumentCaptor<HttpEntity> httpEntityArgumentCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+		when(restTemplate.exchange(eq(url), eq(HttpMethod.POST), httpEntityArgumentCaptor.capture(), 
+				eq(String.class))).thenReturn(response);
+		assertThrows(SdwanResponseException.class, ()->{
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:117]: Don't see getURL() call here same as above test case

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+			sdWanDriver.execute(Constants.LIFECYCLE_OPERATION_DELETE_VCE_STATIC_IP, deploymentLocationProperties, "", "");
+        });
+		
+	}
+	
+	@Test
+	@DisplayName("Testing positive scenario for ExecuteOperation")
+	public void executeOperationTest() {
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:125]: It is not clear what you are testing here. Please add proper Display name content and appropriate tests.

File: src/test/java/com/ibm/sdwan/velocloud/service/impl/KafkaExternalMessagingServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+ 	@DisplayName("Testing positive scenario to Send Delayed Execution Async Response")
+ 	public void sendDelayedExecutionAsyncResponseTest() {
+ 		ExecutionAsyncResponse executionAsyncResponse = new ExecutionAsyncResponse();
+ 		executionAsyncResponse.setRequestId("id");
+ 		SendResult<String, Object> sendResult = mock(SendResult.class);
+ 		ListenableFuture<SendResult<String, String>> responseFuture = mock(ListenableFuture.class);
+
+ 		Mockito.when(kafkaTemplate.send(Mockito.anyString(), Mockito.anyString())).thenReturn(responseFuture);
+ 		kafkaExternalMessagingServiceImpl.sendDelayedExecutionAsyncResponse(executionAsyncResponse,"1",
+ 				Duration.ofMillis(100));
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/service/impl/KafkaExternalMessagingServiceImplTest.java:49]: What you are testing after making senDelayedExecutionAsynResponse() method?

File: src/test/java/com/ibm/sdwan/velocloud/service/impl/KafkaInternalMessagingServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ 	@DisplayName("Testing positive scenario to Send Delayed Execution Async Response")
+    public void  sendEdgeStatusAsyncResponse() {
+    	EdgeStatusMessage edgeStatusMessage = new EdgeStatusMessage();
+    	edgeStatusMessage.setId(123);
+    	SendResult<String, Object> sendResult = mock(SendResult.class);
+ 		ListenableFuture<SendResult<String, String>> responseFuture = mock(ListenableFuture.class);
+
+ 		Mockito.when(kafkaTemplate.send(Mockito.anyString(), Mockito.anyString())).thenReturn(responseFuture);
+    	kafkaInternalMessagingServiceImpl.sendEdgeStatusAsyncResponse(edgeStatusMessage);
+
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/service/impl/KafkaInternalMessagingServiceImplTest.java:47]: Can you add any expectations?

**Review by lokanalla — CHANGES_REQUESTED** [review]
Please rework on the test cases.

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+				eq(String.class))).thenReturn(response);
+		assertThrows(SdwanResponseException.class, ()->{
+			sdWanDriver.execute(Constants.LIFECYCLE_DELETE, deploymentLocationProperties, "", "");
+			
+        });
+		
+	}
+	
+	@Test
+	@DisplayName("Testing positive scenario Execute URL for LIFECYCLE_OPERATION_DELETE_VCE_STATIC_IP")
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:130]: Is it a positive scenario? I see you are asserting Exception.

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+				eq(String.class))).thenReturn(response);
+		assertThrows(SdwanResponseException.class, ()->{
+			sdWanDriver.execute(Constants.LIFECYCLE_OPERATION_DELETE_VCE_STATIC_IP, deploymentLocationProperties, "", "");
+			
+        });
+		
+	}
+	
+	@Test
+	@DisplayName("Testing positive scenario for ExecuteEdge - with exception")
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:146]: This looks positive scenario.

File: src/test/java/com/ibm/sdwan/velocloud/service/impl/KafkaInternalMessagingServiceImplTest.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+   }
+ 
+    @Test
+ 	@DisplayName("Testing exception scenario for Async Response--JsonProcessingException")
+    public void  sendEdgeStatusAsyncResponseexception() throws JsonProcessingException {
+    	SendResult<String, Object> sendResult = mock(SendResult.class);
+ 		ListenableFuture<SendResult<String, String>> responseFuture = mock(ListenableFuture.class);
+
+ 		Mockito.when(objectMapper.writeValueAsString(any())).thenThrow(JsonProcessingException.class);
+ 		kafkaInternalMessagingServiceImpl.sendEdgeStatusAsyncResponse(null);
```
Comment (lokanalla) [inline src/test/java/com/ibm/sdwan/velocloud/service/impl/KafkaInternalMessagingServiceImplTest.java:58]: Where is the exception you are expecting?

**Review by lokanalla — CHANGES_REQUESTED** [review]
Please comment on each comment I made and resolve them. Also still see some missing expectations and the Display content. Please work on them

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java (hunk 9 above)
Comment (siddalasha) [reply src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:130]: it is  Exception  scenario. display name changed to  "Testing exception scenario to Execute URL for LIFECYCLE_OPERATION_DELETE_VCE_STATIC_IP"

File: src/test/java/com/ibm/sdwan/velocloud/service/impl/KafkaInternalMessagingServiceImplTest.java (hunk 11 above)
Comment (siddalasha) [reply src/test/java/com/ibm/sdwan/velocloud/service/impl/KafkaInternalMessagingServiceImplTest.java:58]: by giving Verify method , we can expect the exception

File: src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java (hunk 10 above)
Comment (siddalasha) [reply src/test/java/com/ibm/sdwan/velocloud/driver/SDWanDriverTest.java:146]: yes positive scenario. display name changed to "Testing positive scenario for GET_EDGE"

File: pom.xml [hunk 12]
```diff
@@ -372,6 +373,23 @@
...
                 </executions>
+                <configuration>
+                    <excludes>
+                        <!-- exclude classes in package com.acme.models whose name starts with Spring -->
+                        <exclude>com/ibm/sdwan/velocloud/model/**/*</exclude>
+                        <exclude>com/ibm/sdwan/velocloud/security/**/*</exclude>
+                        <exclude>com/ibm/sdwan/velocloud/config/**/*</exclude>
+                        <exclude>com/ibm/sdwan/velocloud/web/NotImplementedException.class</exclude>
+                        <exclude>com/ibm/sdwan/velocloud/web/SdwanExceptionHandlingControllerAdvice.class</exclude>
+                        <exclude>com/ibm/sdwan/velocloud/utils/LogUtils.class</exclude>
```
Comment (lokanalla) [inline pom.xml:384]: Why have you excluded LogUtils class?

**Review by lokanalla — APPROVED** [review]

**Pull request merged**